charts/argocd-notifications/templates/service-metrics.yaml

```
@@ -5,6 +5,8 @@ metadata:
   name: {{ include "argocd-notifications.name" . }}-metrics
   labels:
     {{- include "argocd-notifications.metrics.labels" . | nindent 4 }}
+  annotations:
+    {{- include "argocd-notifications.metrics.annotations" . | nindent 4 }}
```

Hmm I don't think we should introduce more functions only used once. Please use:
```suggestion
  {{- with .Values.metrics.service.annotations  }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```